pkg/controller/migplan/registry.go

```diff
@@ -97,6 +106,11 @@ func (r ReconcileMigPlan) ensureRegistrySecret(client k8sclient.Client, plan *mi
 		return nil, err
 	}
 	if foundSecret == nil {
+		// if for some reason secret was deleted, we need to make sure we redeploy
+		deleteErr := r.deleteImageRegistryResourcesForClient(client, plan)
```

Comment:
FTR, deletion here is intentionally before creating/updating the secret to avoid weird side effects in the order was the other way around